gum/backend-arm64/gumstalker-arm64.c

```diff
@@ -119,12 +738,1491 @@ gum_stalker_add_call_probe (GumStalker * self,
+      gum_exec_block_write_single_step_transfer_code (block, &gc);
+    }
+
+#if DEBUG_LEVEL>=9
+    {
+      guint8 * begin = block->code_end;
+      block->code_end = gum_arm64_writer_cur (cw);
+      gum_disasm (begin, block->code_end - begin, "\x1b[34mNEW INS> ");
+    }
+#else
+    block->code_end = gum_arm64_writer_cur (cw);
+#endif
+
+    if (gum_exec_block_is_full (block))
+    {
+      gc.continuation_real_address = insn.end;
+#if DEBUG_LEVEL == 10
+      g_print ("gum_exec_ctx_obtain_block_for - block is full\n");
+      g_print ("gc.continuation_real_address: %p", gc.continuation_real_address);
+#endif
+      break;
+    }
+    else if (insn.ci->id == ARM64_INS_BL)
+    {
+      /* We always stop on a call unless it's to an excluded range */
+      if ((requirements & GUM_REQUIRE_RELOCATION) != 0)
+      {
+#if DEBUG_LEVEL == 10
+        g_print ("gum_exec_ctx_obtain_block_for - block is not full\n");
+        g_print ("(requirements & GUM_REQUIRE_RELOCATION) != 0\n");
+        g_print ("rl->eob = FALSE");
+#endif
+        rl->eob = FALSE;
+      }
+      else
+      {
+        break;
+      }
+    }
+    else if (gum_arm64_relocator_eob (rl))
+    {
+      break;
+    }
+  }
+
+  if (gc.continuation_real_address != NULL)
+  {
+    GumBranchTarget continue_target = { 0, };
+
+    continue_target.is_indirect = FALSE;
+    continue_target.absolute_address = gc.continuation_real_address;
+#if DEBUG_LEVEL == 10
+    g_print (
+        "gum_exec_ctx_obtain_block_for - gc.continuation_real_address != NULL\n");
+    g_print ("continue_target.absolute_address: %p",
+        continue_target.absolute_address);
+#endif
+    gum_exec_block_write_jmp_transfer_code (block, &continue_target, &gc);
+  }
+
+  gum_arm64_writer_put_brk_imm (cw, 14);
+
+  gum_arm64_writer_flush (cw);
+
+  block->code_end = (guint8 *) gum_arm64_writer_cur (cw);
+
+  block->real_begin = (guint8 *) rl->input_start;
+  block->real_end = (guint8 *) rl->input_cur;
+
+  gum_exec_block_commit (block);
+
+#if DEBUG_LEVEL == 10
+  g_print ("gum_exec_ctx_obtain_block_for - exit\n");
+  g_print ("\tcode_address_ptr: %p\n", code_address_ptr);
+  g_print ("\tcode_address: %p\n", *code_address_ptr);
+  g_print ("\tblock->code_begin: %p\n", block->code_begin);
+  g_print ("\tblock->code_end: %p\n", block->code_end);
+  g_print ("\tblock->real_begin: %p\n", block->real_begin);
+  g_print ("\tblock->real_end: %p\n", block->real_end);
+#endif
+  return block;
+}
+
+static void
+gum_exec_ctx_write_prolog (GumExecCtx * ctx,
+                           GumPrologType type,
+                           gpointer ip,
+                           GumArm64Writer * cw)
+{
+#if DEBUG_LEVEL == 10
+  g_print ("+ gum_exec_ctx_write_prolog - type: %d\n",type);
+#endif
+
+  /* 1) move to the red-zone */
+  gum_arm64_writer_put_sub_reg_reg_imm (cw, ARM64_REG_SP, ARM64_REG_SP,
+      GUM_RED_ZONE_SIZE);
+
+  /* 2) push registers that are going to be clobbered */
+  int immediate_for_sp = GUM_RED_ZONE_SIZE;
```

Comment:
Declaration after statement. And please use `gint`. Though in this case use `guint` because the value is never negative.